service/lxd.go

```diff
@@ -68,9 +68,16 @@ func (s LXDService) remoteClient(secret string, address string, port int64) (lxd
 		return nil, err
 	}
 
+	serverCert, err := s.m.FileSystem.ServerCert()
+	if err != nil {
+		return nil, err
+	}
+
 	remoteURL := c.URL()
 	client, err := lxd.ConnectLXD(remoteURL.String(), &lxd.ConnectionArgs{
 		HTTPClient:         c.Client.Client,
+		TLSClientCert:      string(serverCert.PublicKey()),
+		TLSClientKey:       string(serverCert.PrivateKey()),
 		InsecureSkipVerify: true,
```

Comment:
We need to set `InsecureSkipVerify` here because this is using the microcloud proxy. We want to bypass the traditional authentication in favour of checking against the auth secret from mDNS.

Reply:
I've made this behave like the other remote clients where we set `InsecureSkipVerify` if we don't have a cluster cert, or if we have an mDNS auth secret.

Reply:
Great, so as we connect to the remote MicroCloud which proxies to the local LXD unix socket, we can use the MicroCloud (microcluster) `cluster.crt` here. Neat.

Reply:
This now looks to cause troubles during connection to the remote, see https://github.com/canonical/microcloud/actions/runs/10086451101/job/27889027120?pr=355#step:13:1240

Reply:
I guess the certificate is valid for the hosts name so do we have to connect through DNS instead of IP?

Reply:
For the sake of closing this carry-over, I've removed this part from this PR and I'll try to see what we can do in a different PR. I think it should be fine because it's not directly related to `service add` and `service list`.

Reply:
Jup that's fine.